runtime/src/main/scala/org/apache/pekko/grpc/internal/ByteStringInputStream.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * license agreements; and to You under the Apache License, version 2.0:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * This file is part of the Apache Pekko project, which was derived from Akka.
+ */
+
+/*
+ * Copyright (C) 2009-2022 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package org.apache.pekko.grpc.internal
+
+import java.io.{ ByteArrayInputStream, InputStream, SequenceInputStream }
+import java.lang.invoke.{ MethodHandles, MethodType }
+
+import scala.util.Try
+
+import org.apache.pekko
+import pekko.annotation.InternalApi
+import pekko.util.ByteString
+import pekko.util.ByteString.ByteString1C
+import pekko.util.ByteString.ByteStrings
+import pekko.util.ccompat.JavaConverters._
+
+/** INTERNAL API */
+@InternalApi
+private[internal] object ByteStringInputStream {
+
+  private val byteStringInputStreamMethodTypeOpt = Try {
+    val lookup = MethodHandles.publicLookup()
+    val inputStreamMethodType = MethodType.methodType(classOf[InputStream])
+    lookup.findVirtual(classOf[ByteString], "asInputStream", inputStreamMethodType)
+  }.toOption
+
+  def apply(bs: ByteString): InputStream = bs match {
+    case cs: ByteString1C =>
+      getInputStreamUnsafe(cs)
+    case _ =>
+      if (byteStringInputStreamMethodTypeOpt.isDefined) {
+        byteStringInputStreamMethodTypeOpt.get.invoke(bs).asInstanceOf[InputStream]
+      } else {
+        legacyConvert(bs)
+      }
+  }
```

Comment:
and what about:

```suggestion
  def apply(bs: ByteString): InputStream = {
    if (byteStringInputStreamMethodTypeOpt.isDefined) {
      byteStringInputStreamMethodTypeOpt.get.invoke(bs).asInstanceOf[InputStream]
    } else {
      legacyConvert(bs)
    }
  }
```

Reply:
another improvement could be the use of `null` instead of `None` in `byteStringInputStreamMethodTypeOpt`

Reply:
I'll check it again but when I was benchmarking the changes before - having the ByteString1C case seemed to help performance. I can't track down the benchmarks that I used, I may need to write them again.

Reply:
actually - the benchmark code that I used a few weeks ago is at https://github.com/pjfanning/pekko-test/

Reply:
With your optimised version of legacyConvert, we may not need the MethodHandle. That code is basically what adInputStream does anyway.

Reply:
I checked in the simpler version. It seems good enough and Pekko 1.0 users benefit with this version.